.github/workflows/codeql.yml

```
@@ -0,0 +1,29 @@
+name: "Code Scanning - Action"
+on:
+  push:
+    branches: [develop]
+  pull_request:
+    branches: [develop]
+jobs:
+  CodeQL-Build:
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: java
+          queries: +security-and-quality
+      - name: Build
+        run: mvn -T1C -DskipTests -DskipChecks install
```

```suggestion
        run: mvn -B -T1C -DskipTests -DskipChecks install
```

We probably want to run this in batch mode (i.e. non interactive)

---

Thanks, didn't know about this one!